applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c

```diff
@@ -4,3 +4,173 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+#if (CONFIG_AUDIO_DEV == GATEWAY)
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/audio/audio.h>
+
+#include "le_audio.h"
+#include "macros_common.h"
+#include "ctrl_events.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(bis_gateway, CONFIG_LOG_BLE_LEVEL);
+
+/* When BROADCAST_ENQUEUE_COUNT > 1 we can enqueue enough buffers to ensure that
+ * the controller is never idle
+ */
+#define BROADCAST_ENQUEUE_COUNT 2U
+#define TOTAL_BUF_NEEDED (BROADCAST_ENQUEUE_COUNT * CONFIG_BT_AUDIO_BROADCAST_SRC_STREAM_COUNT)
+
+BUILD_ASSERT(CONFIG_BT_ISO_TX_BUF_COUNT >= TOTAL_BUF_NEEDED,
+	     "CONFIG_BT_ISO_TX_BUF_COUNT should be at least "
+	     "BROADCAST_ENQUEUE_COUNT * CONFIG_BT_AUDIO_BROADCAST_SRC_STREAM_COUNT");
+
+static struct bt_audio_lc3_preset lc3_preset = BT_AUDIO_LC3_BROADCAST_PRESET_48_4_1;
+// TODO: Currently only 1 stream is supported
+static struct bt_audio_stream streams[CONFIG_BT_AUDIO_BROADCAST_SRC_STREAM_COUNT];
+static struct bt_audio_broadcast_source *broadcast_source;
+
+NET_BUF_POOL_FIXED_DEFINE(tx_pool, TOTAL_BUF_NEEDED, BT_ISO_SDU_BUF_SIZE(CONFIG_BT_ISO_TX_MTU), 8,
+			  NULL);
+
+static uint32_t send_req_cnt;
+static uint32_t sent_cnt;
+static bool stopping;
+
+static void stream_sent_cb(struct bt_audio_stream *stream)
+{
+	int32_t diff;
+
+	sent_cnt++;
+
+	diff = send_req_cnt - sent_cnt;
+
+	// TODO: Use the atomic functionality from ble_trans
+	// if (diff < 0) {
+	// 	/* Should not happen */
+	// 	LOG_WRN("Sending underrun");
+	// } else if (diff > 0) {
+	// 	LOG_WRN("Sending overrun");
+	// }
+
+	if ((sent_cnt % 1000U) == 0U) {
```

Comment:
Maybe remove this statistics, or at least change to DBG

Reply:
The thought here is to implement the same functionality as 'atomic_t iso_tx_pool_alloc' in ble_trans.c to keep track of overruns/underruns